File: Universal Presenter Remote/Universal Presenter Remote WatchKit Extension/DBZ_WKControlView.m
//
//  DBZ_WKControlView.m
//  Universal Presenter Remote
//
//  Created by Brendan Boyle on 3/15/15.
//  Copyright (c) 2015 DBZ Technology. All rights reserved.
//

#import "DBZ_WKControlView.h"
#import "DBZ_ServerCommunication.h"

@interface DBZ_WKControlView()

@end


@implementation DBZ_WKControlView

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    
    // Configure interface objects here.
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
    
    //self.wormhole = [[MMWormhole alloc] initWithApplicationGroupIdentifier:@"group.com.dbztech.Universal-Presenter-Remote.wormhole" optionalDirectory:@"wormhole"];
    
    NSDictionary *requst = @{@"request":@"ConnectSession"};
    
    /*
    [DBZ_WKControlView openParentApplication:requst reply:^(NSDictionary *replyInfo, NSError *error) {
        
        if (error) {
            NSLog(@"%@", error);
        } else {
            
            NSLog(@"%@",[replyInfo objectForKey:@"response"]);
        }
    }];
     
    
    // Become a listener for changes to the wormhole for the button message
    [self.wormhole listenForMessageWithIdentifier:@"UPRWatchAction" listener:^(id messageObject) {
        // The number is identified with the buttonNumber key in the message object
        NSString *action = [messageObject valueForKey:@"action"];
        if ([action  isEqual: @"EndSession"]) {
            [self popToRootController];
        }
    }];
     */
    
    
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
    
    NSDictionary *requst = @{@"request":@"EndSession"};
    
    /*
    [DBZ_WKControlView openParentApplication:requst reply:^(NSDictionary *replyInfo, NSError *error) {
        
        if (error) {
            NSLog(@"%@", error);
        } else {
            
            NSLog(@"%@",[replyInfo objectForKey:@"response"]);
        }
        
    }];
     */
}

- (IBAction)mediaPressed {
    //[self.wormhole passMessageObject:@{@"token" : @(3)} identifier:@"UPRWatchAction"];
    NSDictionary *requst = @{@"request":@"ChangeSlideMedia"};
    
    /*
    [DBZ_WKControlView openParentApplication:requst reply:^(NSDictionary *replyInfo, NSError *error) {
        
        if (error) {
            NSLog(@"%@", error);
        } else {
            
            NSLog(@"%@",[replyInfo objectForKey:@"response"]);
        }
        
    }];
     */
}

- (IBAction)nextPressed {
    NSDictionary *requst = @{@"request":@"ChangeSlideUp"};
    
    /*
    [DBZ_WKControlView openParentApplication:requst reply:^(NSDictionary *replyInfo, NSError *error) {
        
        if (error) {
            NSLog(@"%@", error);
        } else {
            
            NSLog(@"%@",[replyInfo objectForKey:@"response"]);
        }
        
    }];
     */
    
}

- (IBAction)previousPressed {
    NSDictionary *requst = @{@"request":@"ChangeSlideDown"};
    
    /*
    [DBZ_WKControlView openParentApplication:requst reply:^(NSDictionary *replyInfo, NSError *error) {
        
        if (error) {
            NSLog(@"%@", error);
        } else {
            
            NSLog(@"%@",[replyInfo objectForKey:@"response"]);
        }
        
    }];
     */
}

@end

File: Universal Presenter Remote/Universal Presenter Remote WatchKit Extension/DBZ_WKControlView.swift
//
//  DBZ_WKControlView.swift
//  Universal Presenter Remote
//
//  Created by Brendan Boyle on 4/27/16.
//  Copyright © 2016 DBZ Technology. All rights reserved.
//

import WatchKit
import Foundation


class DBZ_WKControlView: WKInterfaceController {

    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        
        // Configure interface objects here.
    }

    override func willActivate() {
        // This method is called when watch view controller is about to be visible to user
        super.willActivate()
    }

    override func didDeactivate() {
        // This method is called when watch view controller is no longer visible
        super.didDeactivate()
    }

    @IBAction func mediaPressed() {
        DBZ_ServerCommunication.getResponse("PlayMedia", withToken: DBZ_ServerCommunication.token(), withHoldfor: true, withDeviceToken: false, withTarget: nil)
    }
    
    @IBAction func nextPressed() {
        DBZ_ServerCommunication.getResponse("SlideUp", withToken: DBZ_ServerCommunication.token(), withHoldfor: true, withDeviceToken: false, withTarget: nil)
    }
    
    @IBAction func previousPressed() {
        DBZ_ServerCommunication.getResponse("SlideDown", withToken: DBZ_ServerCommunication.token(), withHoldfor: true, withDeviceToken: false, withTarget: nil)
    }
}
